src/swap-routers/ISwapRouter.sol

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.13;
+
+interface ISwapRouter {
+    function from() external pure returns (address);
+    function to() external pure returns (address);
+    function swapDefault(uint256 amount, uint256 slippageTolerance) external;
+    function swap0x(bytes calldata swapData, uint256 amount) external payable;
+}
```

Comment:
you don't really need this interface, you can move function definitions to the "BaseSwapRouter" contract and also the name is introducing some confusion with the uniswap's swap router interface